app/models/course/assessment.rb

```diff
@@ -215,4 +225,19 @@ def tab_in_same_course
     return unless tab_id_changed?
     errors.add(:tab, :not_in_same_course) unless tab.category.course == course
   end
+
+  def selected_test_type_for_grading
+    errors.add(:no_test_type_chosen) unless use_public || use_private || use_evaluation
+  end
+
+  # Re-grades all submissions to programming_questions after any change to
+  # test case booleans has been committed
+  def grade_with_new_test_cases
+    return unless (previous_changes.keys.map(&:to_sym) & \
```

Comment:
Does hound complain if you just do this instead?

`return if (previous_changes.keys & ['use_private', 'use_public', 'use_evaluation']).empty?`